rust-runtime/smithy-json/src/escape.rs

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use std::borrow::Cow;
+
+const ESCAPES: &[char] = &['"', '\\', '\u{08}', '\u{0C}', '\n', '\r', '\t'];
+
+/// Escapes a string for embedding in a JSON string value.
+pub fn escape_string(value: &str) -> Cow<str> {
+    if !value.contains(ESCAPES) {
+        return Cow::Borrowed(value);
+    }
+
+    let mut escaped = String::new();
+    let (mut last, end) = (0, value.len());
+    for (index, chr) in value
+        .char_indices()
+        .filter(|(_index, chr)| ESCAPES.contains(chr))
+    {
+        escaped.push_str(&value[last..index]);
+        escaped.push_str(match chr {
+            '"' => "\\\"",
+            '\\' => "\\\\",
+            '\u{08}' => "\\b",
+            '\u{0C}' => "\\f",
+            '\n' => "\\n",
+            '\r' => "\\r",
+            '\t' => "\\t",
+            _ => unreachable!(),
+        });
+        last = index + 1;
+    }
+    escaped.push_str(&value[last..end]);
```

Comment:
can you add a test where there is data after the escape? As it is currently, the tests will pass if this line is deleted